CONTRIBUTING.md

```diff
@@ -16,3 +18,36 @@ Prometheus uses GitHub to manage reviews of pull requests.
   and the _Formatting and style_ section of Peter Bourgon's [Go: Best
   Practices for Production
   Environments](http://peter.bourgon.org/go-in-production/#formatting-and-style).
+
+
+## Steps to Contribute
+
+Should you wish to work on an issue, please claim it first by commenting on the GitHub issue that you want to work on it. This is to prevent duplicated efforts from contributors on the same issue.
+
+Please check the [`low-hanging-fruit`](https://github.com/prometheus/prometheus/issues?q=is%3Aissue+is%3Aopen+label%3A%22low+hanging+fruit%22) label to find issues that are good for getting started. If you have questions about one of the issues, with or without the tag, please comment on them and one of the maintainers will clarify it. For a quicker response, contact us over [IRC](https://prometheus.io/community).
+
+For complete instructions on how to compile see: [Building From Source](https://github.com/prometheus/prometheus#building-from-source)
+
+For quickly compiling and testing your changes do:
+```
+# For building.
+go build ./cmd/prometheus/
+./prometheus
+
+# For testing.
+make test-short   # To quickly test.
+
+make test         # For more thorough testing before you commit and push.
+```
+
+All our issues are regularly tagged so that you can also filter down the issues involving the components you want to work on. For our labelling policy refer [the wiki page](https://github.com/prometheus/prometheus/wiki/Label-Names-and-Descriptions).
+
+## Pull Request Checklist
+
+* Branch from the master branch and, if needed, rebase to the current master branch before submitting your pull request. If it doesn't merge cleanly with master you may be asked to rebase your changes.
+
+* Commits should be as small as possible, while ensuring that each commit is correct independently (i.e., each commit should compile and pass tests).
```

Comment:
"as small as possible" could lead to undesired results. E.g. when a helper function is added for something I don't want it to be its own commit.
But likely we have more problems with too long commits. We can go with this but keep in mind to adjust it in case we get a bunch of 10+ commit PRs.